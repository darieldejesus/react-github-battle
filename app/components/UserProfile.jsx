import React, { Component } from 'react';
import PropTypes from 'prop-types';

import PlayerPreview from './PlayerPreview';
import UserProperties from './UserProperties';

/**
 * Shows the main properties of a username
 * including an image.
 */
const UserProfile = ({ profile }) => (
  <PlayerPreview
    username={profile.login}
    avatar={profile.avatar_url}>
    <UserProperties profile={profile} />
  </PlayerPreview>
)

UserProfile.propTypes = {
  profile: PropTypes.shape({
    avatar_url: PropTypes.string.isRequired,
    followers: PropTypes.number.isRequired,
    following: PropTypes.number.isRequired,
    login: PropTypes.string.isRequired,
    public_repos: PropTypes.number.isRequired,
    blog: PropTypes.string,
    company: PropTypes.string,
    location: PropTypes.string,
    name: PropTypes.string
  })
}

export default UserProfile;
