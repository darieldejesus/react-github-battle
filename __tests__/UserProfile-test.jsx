import React from 'react';
import { shallow } from 'enzyme';

import UserProfile from '../app/components/UserProfile';

describe('<UserProfile />', () => {
  it('User Profile is rendered with the required props', () => {
    const profile = {
      login: 'username',
      avatar_url: 'https://example.com/username.jpg',
      followers: 10,
      following: 5,
      public_repos: 3
    };
    const userProfile = shallow(
      <UserProfile profile={profile} />
    );
    expect(userProfile).toMatchSnapshot();
  });
});