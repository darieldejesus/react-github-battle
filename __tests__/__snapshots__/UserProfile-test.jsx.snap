// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserProfile /> User Profile is rendered with the required props 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <PlayerPreview
    avatar="https://example.com/username.jpg"
    username="username"
  >
    <UserProperties
      profile={
        Object {
          "avatar_url": "https://example.com/username.jpg",
          "followers": 10,
          "following": 5,
          "login": "username",
          "public_repos": 3,
        }
      }
    />
  </PlayerPreview>,
  "nodes": Array [
    <PlayerPreview
      avatar="https://example.com/username.jpg"
      username="username"
    >
      <UserProperties
        profile={
          Object {
            "avatar_url": "https://example.com/username.jpg",
            "followers": 10,
            "following": 5,
            "login": "username",
            "public_repos": 3,
          }
        }
      />
    </PlayerPreview>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <UserProfile
        profile={
          Object {
            "avatar_url": "https://example.com/username.jpg",
            "followers": 10,
            "following": 5,
            "login": "username",
            "public_repos": 3,
          }
        }
      />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "profile": Object {
            "avatar_url": "https://example.com/username.jpg",
            "followers": 10,
            "following": 5,
            "login": "username",
            "public_repos": 3,
          },
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <PlayerPreview
          avatar="https://example.com/username.jpg"
          username="username"
        >
          <UserProperties
            profile={
              Object {
                "avatar_url": "https://example.com/username.jpg",
                "followers": 10,
                "following": 5,
                "login": "username",
                "public_repos": 3,
              }
            }
          />
        </PlayerPreview>,
        "_debugID": 2,
        "_renderedOutput": <PlayerPreview
          avatar="https://example.com/username.jpg"
          username="username"
        >
          <UserProperties
            profile={
              Object {
                "avatar_url": "https://example.com/username.jpg",
                "followers": 10,
                "following": 5,
                "login": "username",
                "public_repos": 3,
              }
            }
          />
        </PlayerPreview>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <UserProfile
    profile={
      Object {
        "avatar_url": "https://example.com/username.jpg",
        "followers": 10,
        "following": 5,
        "login": "username",
        "public_repos": 3,
      }
    }
  />,
}
`;
